### PR TITLE
[aks-preview] Fix bug related to updating DCR when non monitoring add enabled through az aks enable-addons 

### PR DESCRIPTION
**Pull request by ganga1980**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

Fix will avoid updating existing data collection rules if non monitoring addon trying to enable through
az aks enable-addons 
### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by zhoxing-ms** [comment]
@ganga1980 Please add some test cases for this change and fix these CI issues

**Comment by FumingZhang** [comment]
CI failed on test case `test_aks_create_and_update_with_http_proxy_config`, fixing in #8170

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=106255113&view=results) to validate the change.

- test_aks_enable_monitoring_with_aad_auth_msi
- test_aks_enable_monitoring_with_aad_auth_msi_with_syslog
- test_aks_addon_enable_monitoring_with_aad_auth_uai_with_syslog

**Comment by ganga1980** [comment]
> @ganga1980 Please add some test cases for this change and fix these CI issues

Thanks, @FumingZhang  for fixing test failure.

**Review by FumingZhang — APPROVED** [review]

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -9,6 +9,10 @@ If there is no rush to release a new version, please just add a description of t
 
 To release a new version, please select a new version number (usually plus 1 to last patch version, X.Y.Z -> Major.Minor.Patch, more details in `\doc <https://semver.org/>`_), and then add a new section named as the new version number in this file, the content should include the new modifications and everything from the *Pending* section. Finally, update the `VERSION` variable in `setup.py` with this new version number.
 
+9.0.0b7
++++++++
+* Fix bug related to updating the monitoring addon DCR when the non monitoring addon enabled through az aks enable-addons.
```
Comment (yanzhudd) [inline src/aks-preview/HISTORY.rst:14]:
```suggestion
* Fix bug related to updating the monitoring addon DCR when the non monitoring addon enabled through `az aks enable-addons`.
```

**Comment by yanzhudd** [comment]
please add some test cases for the bug to verify the changes.

**Comment by ganga1980** [comment]
> please add some test cases for the bug to verify the changes.

added test to cover.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**